### PR TITLE
feat: scheduler csv

### PR DESCRIPTION
**Pull request by rephus**
Closes: #4452

### Description:

Email chart CSV
image

Email dashboard CSV 

Screenshot from 2023-02-17 11-33-42

Slack chart CSV

Screenshot from 2023-02-17 12-28-27

Slack dashboard CSV

image

### Acceptance criteria:

Miro sketch here: https://miro.com/app/board/uXjVP1D_L4w=/?moveToWidget=3458764546073253136&cot=14

- [x] I can choose the format for my scheduled delivery (either `image` or `CSV`). This option is available in both Slack and email. 
- [x] If I select `CSV` format, then my delivery has the results from the charts included as .csv files in my scheduled delivery. 
- [ ] Each chart has its own .csv file attachment (if sending a dashboard)
- [ ] By default, the results included in the .csv file are whatever results are shown in the chart. E.g. if I opened the chart in `view` mode, my .csv file should look like the `results` table.
- [ ] By default, the results tables are sent in the format that they appear in the results table. But, users can change this in the `advanced settings` tab.
- [ ] I can change the limit of values shared in my .csv file and the format they appear in (just like downloading a .csv from the results table).
- [ ] If I am sending a dashboard in `CSV` format, I can only select one set of options for the csv results that will be applied to _**all**_ of the charts (I can't set up if the csv is formatted vs. raw, for example, for each of my charts independently).

### CODE REVIEW

**Comment by rephus** [comment]
@TuringLovesDeathMetal maybe we could add `add csv as attachment` on emails as an advanced option ?

**Comment by TuringLovesDeathMetal** [comment]
@rephus let's keep them as links for now, and then if people request attachments, we can consider them!

**Comment by rephus** [comment]
I've added email (dev) , slack and S3 to the render PR env

**Comment by rephus** [comment]
Invite e2e test failing, ignoring 

image

**Comment by rephus** [comment]
CLI tests are currently failing on master, https://github.com/lightdash/lightdash/pull/4525 and it was working fine until I mergeed main here, so we can ignore...

File: packages/backend/src/scheduler/SchedulerTask.ts [hunk 1]
```diff
@@ -47,6 +67,8 @@ const getChartOrDashboard = async (
     throw new Error("Chart or dashboard can't be both undefined");
 };
 
+// getCsvForChart('5de7b22b-813d-46a9-b093-ff2d83912253')
```
Comment (ZeRego) [inline packages/backend/src/scheduler/SchedulerTask.ts:70]: forgotten comment

**Review by ZeRego — APPROVED** [review]

**Pull request merged**